play_store_export/export.py

```diff
@@ -0,0 +1,195 @@
+) -> List[bigquery_datatransfer.types.TransferRun]:
+    """
+    Start backfill of the given date range and return list of tuples of name and scheduled time
+    for transfer runs created (1 per day)
+    """
+    logging.info(f"Starting backfill {start_date} to {end_date}")
+
+    date_diff = (end_date - start_date).days
+    if date_diff < 0:
+        raise ValueError("End date must be equal to or greater than start date")
+    if (end_date - start_date).days >= BACKFILL_DAYS_MAX:
+        raise ValueError(f"A maximum of {BACKFILL_DAYS_MAX} days is allowed per backfill request")
+
+    # end_date is midnight of the day after the last day
+    end_date = to_utc_midnight_datetime(end_date) + datetime.timedelta(days=1)
+    start_date = to_utc_midnight_datetime(start_date)
+
+    response = client.start_manual_transfer_runs(
+        parent=transfer_config_name,
+        requested_time_range={
+            "start_time": bigquery_datatransfer.types.Timestamp(
+                seconds=int(start_date.timestamp())),
+            "end_time": bigquery_datatransfer.types.Timestamp(
+                seconds=int(end_date.timestamp())),
+        },
+    )
+    return response.runs
+
+
+def wait_for_transfer(transfer_name: str, timeout: int = 1200, polling_period: int = 20) -> int:
+    """
+    Continuously poll for run status to wait for completion
+    """
+    client = bigquery_datatransfer.DataTransferServiceClient()
+
+    state = transfer_enums.TransferState.PENDING
+
+    time_elapsed = 0
+    while (state == transfer_enums.TransferState.PENDING or
+           state == transfer_enums.TransferState.RUNNING):
+        try:
+            transfer_run = client.get_transfer_run(transfer_name)
+        except GoogleAPICallError as e:
+            # grpc errors are not serializable and cannot be raised in multiprocessing
+            raise DataTransferException(f"Error getting transfer run: {e.message}")
+
+        run_date = datetime.datetime.utcfromtimestamp(transfer_run.run_time.seconds).date()
+
+        state = transfer_run.state
+
+        if not (state == transfer_enums.TransferState.PENDING or
+                state == transfer_enums.TransferState.RUNNING):
+            break
+
+        if time_elapsed >= timeout:
+            logging.info(f"Transfer for {run_date} did not complete in {timeout} seconds")
+            return -1
+        time.sleep(polling_period)
+        time_elapsed += polling_period
+
+    if state == transfer_enums.TransferState.SUCCEEDED:
+        result = "succeeded"
+    elif state == transfer_enums.TransferState.CANCELLED:
+        result = "cancelled"
+    elif state == transfer_enums.TransferState.FAILED:
+        result = "failed"
+    else:
+        result = "unspecified"
+
+    logging.info(f"Transfer for {run_date} {result}")
+
+    return state
+
+
+def start_export(project: str, transfer_config_name: str, transfer_location: str,
+                 base_date: datetime.date, backfill_day_count: int = 35):
```

Comment:
nit: 35 is defined here and below. I would remove it from here and let it solely be defined by the argparser.